src/jsifier.js

```diff
@@ -241,7 +241,7 @@ function JSify(data, functionsOnly) {
         EXPORTED_FUNCTIONS[finalName] = 1;
         Functions.libraryFunctions[finalName] = 2;
       }
-      if ((EXPORT_ALL || (finalName in EXPORTED_FUNCTIONS)) && !noExport) {
+      if (EXPORT_ALL && !(finalName in EXPORTED_FUNCTIONS) && !noExport) {
```

Comment:
Yes, correct assignment is done in emscripten.py. There are 2 similar places:
https://github.com/kripken/emscripten/blob/5387c1e5f1f55b69c41b94cf044062c59e052f0b/emscripten.py#L889-L893
https://github.com/kripken/emscripten/blob/5387c1e5f1f55b69c41b94cf044062c59e052f0b/emscripten.py#L1610-L1614

When the lines generated by the code moved by this PR is executed, variables on the right side from `=` are `undefined` anyway and actually defined later by linked code from emscripten.py mentioned above.

I'm not sure, however, that this code should be moved to emscripten.py, since `!(finalName in EXPORTED_FUNCTIONS)` functions are generated earlier in the process (haven't checked when and where exactly). Basically this function loops through many functions, both explicitly exported by user and those used internally. It is beyond my understanding of the project at the moment.